### PR TITLE
fs/procfs: Change FS_PROCFS_EXCLUDE_xxx default value to DEFAULT_SMALL

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- fs/procfs: Fix warning when all CONFIG_FS_PROCFS_EXCLUDE_xxx are enabled
- fs/procfs: Change FS_PROCFS_EXCLUDE_xxx default value to DEFAULT_SMALL 
- boards: Refresh defconfig after the default value of FS_PROCFS_EXCLUDE_xxx

## Impact
procfs default setting

## Testing
Pass CI

### CODE REVIEW

File: fs/procfs/Kconfig [hunk 1]
```diff
@@ -31,15 +31,15 @@ menu "Exclude individual procfs entries"
 
 config FS_PROCFS_EXCLUDE_PROCESS
 	bool "Exclude process information"
-	default n
+	default DEFAULT_SMALL
```
Comment (acassis) [inline fs/procfs/Kconfig:34]: Hi @xiaoxiang781216 I saw that sched/Kconfig and fs/Kconfig has this already, but I suggest using "default y DEFAULT_SMALL" for legibility. I will submit a patch to fix sched and fs

Comment (xiaoxiang781216) [reply fs/procfs/Kconfig:34]: But, I think `default DEFAULT_SMALL` is more clear than `default y if DEFAULT_SMALL`. For example, what's default value if DEFAULT_SMALL disable?

File: fs/procfs/fs_procfs.c [hunk 2]
```diff
@@ -184,27 +184,24 @@ static const struct procfs_entry_s g_procfs_entries[] =
 #if defined(CONFIG_DEBUG_TCBINFO) && !defined(CONFIG_FS_PROCFS_EXCLUDE_TCBINFO)
   { "tcbinfo",       &tcbinfo_operations,         PROCFS_FILE_TYPE   },
 #endif
+  { NULL,            NULL,                        PROCFS_UNKOWN_TYPE },
 };
 
 #ifdef CONFIG_FS_PROCFS_REGISTER
 static const uint8_t g_base_entrycount = sizeof(g_base_entries) /
```
Comment (pkarashchenko) [inline fs/procfs/fs_procfs.c:191]: why do we need `g_base_entrycount` if array us already NULL terminated and can be just `while`-ed till `pathpattern != NULL`?

Comment (xiaoxiang781216) [reply fs/procfs/fs_procfs.c:191]: Adding NULL entry avoid the out of boundary error:
```
procfs/fs_procfs.c: In function 'procfs_readdir':
Error: procfs/fs_procfs.c:878:35: error: array subscript <unknown> is outside array bounds of 'const struct procfs_entry_s[0]' [-Werror=array-bounds]
  878 |       if (strncmp(g_procfs_entries[level1->base.index].pathpattern,
```
Many place verify the index in the range, the code has to recompute every time without `g_base_entrycount`.

File: fs/procfs/Kconfig (hunk 1 above)
Comment (acassis) [reply fs/procfs/Kconfig:34]: I think the opposite, ```default DEFAULT_SMALL``` give the impression that it is setting DEFAULT_SMALL. The ```default n``` is implicit when DEFAULT_SMALL is not enabled.

File: fs/procfs/fs_procfs.c (hunk 2 above)
Comment (pkarashchenko) [reply fs/procfs/fs_procfs.c:191]: Now we have a similar error, but at line 740

File: fs/procfs/fs_procfs.c [hunk 3]
```diff
@@ -186,25 +176,21 @@ static const struct procfs_entry_s g_procfs_entries[] =
...
                                          sizeof(struct procfs_entry_s);
 
+#ifdef CONFIG_FS_PROCFS_REGISTER
 static FAR struct procfs_entry_s *g_procfs_entries;
 static uint8_t g_procfs_entrycount;
 #else
-static const uint8_t g_procfs_entrycount = sizeof(g_procfs_entries) /
-                                           sizeof(struct procfs_entry_s);
+static FAR const struct procfs_entry_s *g_procfs_entries = g_base_entries;
```
Comment (pkarashchenko) [inline fs/procfs/fs_procfs.c:186]: Maybe we can use `define` and not introduce new variable?

Comment (xiaoxiang781216) [reply fs/procfs/fs_procfs.c:184]: Done.

**Review by acassis — APPROVED** [review]

File: fs/procfs/fs_procfs.c [hunk 4]
```diff
@@ -375,15 +363,14 @@ static int procfs_open(FAR struct file *filep, FAR const char *relpath,
...
+                      g_procfs_entries[x].ops->open);
 
           ret = g_procfs_entries[x].ops->open(filep, relpath, oflags, mode);
-
           if (ret == OK)
             {
               DEBUGASSERT(filep->f_priv);
 
-              ((struct procfs_file_s *) filep->f_priv)->procfsentry =
+              ((struct procfs_file_s *)filep->f_priv)->procfsentry =
```
Comment (pkarashchenko) [inline fs/procfs/fs_procfs.c:373]: Note for one of next PRs.
```suggestion
              ((FAR struct procfs_file_s *)filep->f_priv)->procfsentry =
```

File: fs/procfs/fs_procfs.c [hunk 5]
```diff
@@ -947,7 +911,7 @@ static int procfs_rewinddir(struct inode *mountpt, struct fs_dirent_s *dir)
 
   if (priv->level > 0 && priv->procfsentry == NULL)
     {
-      priv->index = ((struct procfs_level1_s *) priv)->firstindex;
+      priv->index = ((struct procfs_level1_s *)priv)->firstindex;
```
Comment (pkarashchenko) [inline fs/procfs/fs_procfs.c:914]: ditto
```suggestion
      priv->index = ((FAR struct procfs_level1_s *)priv)->firstindex;
```

File: fs/procfs/fs_procfs.c [hunk 6]
```diff
@@ -1025,7 +989,7 @@ static int procfs_statfs(struct inode *mountpt, struct statfs *buf)
 static int procfs_stat(struct inode *mountpt, const char *relpath,
                        struct stat *buf)
```
Comment (pkarashchenko) [inline fs/procfs/fs_procfs.c:990]: ditto
```suggestion
static int procfs_stat(FAR struct inode *mountpt, FAR const char *relpath,
                       FAR struct stat *buf)
```

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@acassis @pkarashchenko done.

**Review by pkarashchenko — APPROVED** [review]

**Review by masayuki2009 — CHANGES_REQUESTED** [review]
@xiaoxiang781216 

Hmm, ps command not found with `hifive1-revb:nsh`

```
NuttShell (NSH) NuttX-3.6.1
nsh> uname -a
NuttX 3.6.1 25df009f87 Oct 17 2022 11:55:14 risc-v hifive1-revb
nsh> help
help usage:  help [-v] [<cmd>]

  ?       cat     echo    help    kill    ls      mount   sleep   uname   usleep

Builtin Apps:
  buttons   sh        getprime  hello     nsh
nsh> ps
nsh: ps: command not found
nsh> Traceback (most recent call last):
```

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216
> 
> Hmm, ps command not found with `hifive1-revb:nsh`
> 
> ```
> NuttShell (NSH) NuttX-3.6.1
> nsh> uname -a
> NuttX 3.6.1 25df009f87 Oct 17 2022 11:55:14 risc-v hifive1-revb
> nsh> help
> help usage:  help [-v] [<cmd>]
> 
>   ?       cat     echo    help    kill    ls      mount   sleep   uname   usleep
> 
> Builtin Apps:
>   buttons   sh        getprime  hello     nsh
> nsh> ps
> nsh: ps: command not found
> nsh> Traceback (most recent call last):
> ```

@masayuki2009 this is because the default value is changed when DEFAULT_SMALL is enabled. I have update all defconfig which enable DEFAULT_SMALL to keep the expanded result same as before. Please try again.

**Comment by masayuki2009** [comment]
@xiaoxiang781216 
I think most of the changes in the first commit relate to style changes. 
So it would be better to separate the commit to style changes and fix warnings.

**Comment by xiaoxiang781216** [comment]
Done.

File: fs/procfs/fs_procfs.c [hunk 7]
```diff
@@ -858,14 +843,16 @@ static int procfs_readdir(FAR struct inode *mountpt,
...
+          strncmp(g_procfs_entries[level1->base.index].pathpattern,
                   g_procfs_entries[level1->firstindex].pathpattern,
                   level1->subdirlen) == 0)
         {
           /* This entry matches.  Report the subdir entry */
 
-          name = &g_procfs_entries[level1->base.index].pathpattern[
-                    level1->subdirlen + 1];
+          name = &g_procfs_entries[level1->base.index].
+                    pathpattern[level1->subdirlen + 1];
```
Comment (masayuki2009) [inline fs/procfs/fs_procfs.c:855]: @xiaoxiang781216 
I think this is a style change, so it would be better to merge the change into the first commit.

Comment (xiaoxiang781216) [reply fs/procfs/fs_procfs.c:855]: Done.

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**